src/domain/fee/feeDistributionService.ts

```
@@ -753,12 +756,14 @@ export async function referralRewardsCalls({
       affiliateAmounts,
       batchSize,
       async (currentBatch: [string, BigNumber][]) => {
-        const accounts = currentBatch.map((item) => item[0]);
-        const amounts = currentBatch.map((item) => item[1]);
+        const params = currentBatch.map(([account, amount]) => ({
```

Can accounts be duplicated (I think normally they would not be duplicated in a certain batch)? Should a duplication check be added when generating the affiliated accounts

---

Q:  should discounts still be included in totals when there is no account ?
i.e.  L719 --> allDiscountUsd = allDiscountUsd.add(discountUsdBn);

---

In theory, I think an account could earn both as an affiliateAccount and also as a discountAccount but that would be valid and I don't think any check would be needed. 

I think after generating the affiliates accounts data in getDistributionData, the account is unique within the list of affiliates (or referrals) given the data is aggregated per account using affiliatesStats.reduce (or referralStats.reduce), thus such a check shouldn't be necessary.

---

I don't think it's possible for account to actually be zero address for discounts or affiliates, but it could make sense to update the ZeroAddress check to throw an error instead of 'continue' in the case that it does happen, as maybe that could imply issues with the underlying subgraph data.

---

I've updated in the latest commit to throw an error in the case of ZeroAddress as it shouldn't happen and would be a sign that something is wrong with the subgraph data.